### ISSUE TITLE
[Feature Request] Automatically download Sketch images when downloading Pixiv profile

### ISSUE THREAD

**Issue by HydroHydro** [Nandaka/PixivUtil2#793]
### Prerequisites

* [x] Did you read FAQ section?
* [x] Did you test with the latest [releases](https://github.com/Nandaka/PixivUtil2/releases "latest releases") or [commit](https://github.com/Nandaka/PixivUtil2/commits/master "latest commit") ?

### Description

Now that we have Pixiv Sketch support, it would be nice if PixivUtil2 could detect the Sketch profile when downloading the main Pixiv profile.

For example, when downloading memberid 711569 in PixivUtil2, it automatically detects the Sketch profile (camori) and downloads **all** images from **all** profiles.

And vice versa. Automatically detect and download the main Pixiv profile when selecting the Sketch profile.

Maybe add this as an option in the config?

**Comment by Nandaka** [comment]
well, I can add additional input when asking the member id whether to download the sketch or not, but the execution will be only after the whole images is downloaded.

https://github.com/Nandaka/PixivUtil2/blob/master/PixivUtil2.py#L171
https://github.com/Nandaka/PixivUtil2/blob/master/PixivUtil2.py#L708

Another option is to implement new batch job type to support Sketch and you just set the batch.json file.

https://github.com/Nandaka/PixivUtil2/blob/master/PixivBatchHandler.py#L203

**Comment by HydroHydro** [comment]
I think it would easier to just add the option in config.
DownloadPixivAndSketch = True
That way it **automatically** downloads images from both Pixiv and Sketch.

Your first idea would put an extra step in the way for people who only want to download one memberid/sketchprofile.

And your second idea... I have no idea what setting a batch file means...

**Comment by Nandaka** [comment]
see https://github.com/Nandaka/PixivUtil2/wiki/Using-Batch-Job-(Experimental)

some people might not want to always download from the Sketch for every member ID, as it will increase the download time.

Are you using `4. Download from list` or `1. Download by member_id`? For option 1, you need to key in the member id anyway, so additional input should not be a problem.

**Comment by Nandaka** [comment]
woops wrong issue number :P

**Comment by HydroHydro** [comment]
A batch job seems too complicated for a simple thing like automatically downloading the Pixiv *and* Sketch images.

> some people might not want to always download from the Sketch for every member ID, as it will increase the download time.
That's why I say add an option to config for people who want **all** images from both profiles! That way the people who want Pixiv *and* Sketch images can just choose it in the config. The current user would be unaffected. No extra input for them.

> Are you using 4. Download from list or 1. Download by member_id? For option 1, you need to key in the member id anyway, so additional input should not be a problem.
Additional input is always a problem. No one wants to have to wait until *after* the Pixiv profile is done downloading to input another command for the same artist! In fact, no one wants additional input **before** either. It's so much easier with a config setting.

And lists... If I want to update my downloaded artists from a list, it would be **terrible** if I had to check it every few minutes to input that I want the Sketch profile for every single one of them!!!!

**Comment by bluerthanever** [comment]
OR, a new menu to download sketches with "list.txt"? Haha. Just saying... Not sure if there are creators in my following list using sketches yet...

**Comment by HydroHydro** [comment]
@bluerthanever I agree that they should not be mixed. For the average user, things should continue as they are right now.
But for the few among us, like myself, who'd prefer to have *all* the images from *both* sites, a simple option in config easily does the job. ;)

A whole new menu or manually making a list.txt file just for sketches is... just making things more complicated than they need to be.

**Comment by Nandaka** [comment]
> 2\. amongst all the creators a user followings, there's gotta be some that post good stuff on sketch side, and some who post poor quality stuff, or let's say contents that don't qualify to be downloaded? So the user may not want to download all contents on sketch side while doing regular download routines.

that's why I prefer to make it as input param when you doing download by member id. If it is a global config, then you need to manually update the config file again when you want to switch modes.

**Comment by HydroHydro** [comment]
> that's why I prefer to make it as input param when you doing download by member id. If it is a global config, then you need to manually update the config file again when you want to switch modes.

For me, if it's in the config file, I will *never* switch back to "only Pixiv mode." I want *all* the images no matter what.

How am I supposed to update my collection if I have to constantly check for entering the new input?

If it **must** be an input, can it be "automatically download Pixiv *and* Sketch images for this entire session"? Rather than me having to enter an input for every profile downloaded.

**Comment by HydroHydro** [comment]
Or.... why not both?

In config, make "DownloadPixivAndSketch = False" the default. This enables the extra input.

"DownloadPixivAndSketch = True" disables the input and automatically downloads both profiles with no extra input required.

**Comment by Nandaka** [comment]
it complicates the code...

**Comment by Nandaka** [comment]
implement using input param, need to test.

**Comment by Nandaka** [comment]
if the artist doesn't have sketch page, it will throw 404 error and it will still continue processing next artists.

**Comment by HydroHydro** [comment]
What about lists? Do I need to enter the extra input for every artist now? :(

**Comment by Nandaka** [comment]
nope, only option 1 implemented. First argument is the y/n flag for pixiv sketch if you run it from cmd/console.

**Comment by ItaloKnox** [comment]
I read over this issue because I suddenly started getting Pixiv Sketch images from my batch jobs (more about that later). Just a simple question I didn't manage to answer myself by reading: how do you actually skip Pixiv Sketch images when running option 1?

I can see this:
> nope, only option 1 implemented. First argument is the y/n flag for pixiv sketch if you run it from cmd/console.

But it feels unclear to me. Is it ``pixivutil2.py --startaction=1 n member_id``? Can't really find anything in the docs to indicate that.

Just to be clear, I wrote my own Python script to invoke and split multiple PixivUtil2 jobs to be more efficient when going through a big list. My script runs multiples instances of PixivUtil2 with the arguments ``'-n', '1', '-x', '--startaction=1'`` (in this order) followed by the split list of artists I want to batch download. That was written long before there was batch support for PixivUtil2, so I'm unaware of how it works. That's just my own personal implementation on a separate script.

**Comment by Nandaka** [comment]
@ItaloKnox  `pixivutil2.py --startaction=1 n member_id` and actually you can remove `n` as the default is not to download the sketch

https://github.com/Nandaka/PixivUtil2/blob/master/PixivUtil2.py#L180

ideally need to restructure the option parser to use keyword, rather than positional args

**Comment by ItaloKnox** [comment]
> 
> 
> @ItaloKnox `pixivutil2.py --startaction=1 n member_id` and actually you can remove `n` as the default is not to download the sketch
> 
> [https://github.com/Nandaka/PixivUtil2/blob/master/PixivUtil2.py#L180](https://github.com/Nandaka/PixivUtil2/blob/master/PixivUtil2.py?rgh-link-date=2020-09-29T01%3A10%3A17Z#L180)
> 
> ideally need to restructure the option parser to use keyword, rather than positional args

Adding ``n`` after ``--startaction=1`` seems to do the trick in my script. Not adding it still downloads the sketch images for some reason. Thanks, that should fix it.

**Comment by jwshields** [comment]
Not sure if this would need a new issue or not, but I've found a case where the Pixiv UID differs from the UID on Sketch...
This seems to cause some oddities when using the `%member_id%` parameter in the Filename config section.
Example, Pixiv: 41141316 / Sketch: 37963593


![image](https://user-images.githubusercontent.com/7400556/95672758-4b1ab080-0b58-11eb-87c9-63b618444145.png)

**Comment by Nandaka** [comment]
can you open as separate issue?
